torch/utils/data/datasets/readfilesfromtardataset.py

```diff
@@ -0,0 +1,58 @@
+from torch.utils.data.dataset import IterableDataset
+from torch.utils.data.datasets.common import validate_pathname_binary_tuple
+from typing import Iterable, Iterator
+
+import os
+import tarfile
+import warnings
+
+class ReadFilesFromTarIterableDataset(IterableDataset):
+    r""" :class:`ReadFilesFromTarIterableDataset`.
+
+    IterableDataset to extract tar binary streams from input iterables
+    yield pathname and extracted binary stream in a tuple.
+    args:
+        dataset: Iterable dataset that provides pathname and tar binary stream in tuples
+        length: a nominal length of the dataset
+    """
+    def __init__(
+            self,
+            dataset : Iterable,
+            length : int = -1):
+        super().__init__()
+        self.dataset : Iterable = dataset
```

Comment:
Please make Iterable typed here: Iterable[tuple[T, T]] and you can drop entire `validate_pathname_binary_tuple`

Reply:
will switch typed Iterable but keep validate_pathname_binary_tuple, since type annotation won't really block the execution.